### DISCUSSION TITLE
LCZ classification around weather station

### DISCUSSION THREAD

**Discussion by Jukjuk-boop** [orbisgis/geoclimate#961] [Q&A]
I'm studying the impact of urban typo-morphology on temperatures. The aim is to determine whether the different LCZ classes present significant temperature differences.
To do this, I assign each station an LCZ class using several methods:
- LCZ at weather station location
- Majority LCZ within a 250m radius of the weather station

I'd like to develop a third method, which is the subject of this topic:
Define the 250m radius around weather stations as a single RSU and then determine the LCZ class according to the indicators present in this area.

I want to point out that I work with BD-TopoV2, BD-TopoV3 and OSM.

Feel free to ask me more questions if I'm unclear about what I want to do.

**Comment by j3r3m1** [comment]
Hi @Jukjuk-boop and thank you for this topic. Your question is super clear, we will try to make the answer as clear...

Currently, you have probably used the command line to run the whole GeoClimate workflow. This workflow is used to prepare the input data (building, vegetation, water, etc.), calculate spatial units (blocks and RSU), calculate indicators for each spatial unit type and then assign to each RSU a LCZ type.

There is currently no option in the workflow to use your own RSU (circle around each station instead of the current "Topographical Spatial Unit"). However, it is possible to run only some part of the workflow described previously. Two possibilities exist, both of them needing the use of groovy scripts.

1. The simplest but less accurate: the idea would be to create your own RSU (being your circle of radius 250 m) and then use this layer as input of the function "computeAllGeoIndicators" of the class WorkflowGeoindicators. However, the current limit of this method is that a building can only be included in a single RSU (circle in your case), thus if two stations are close to each other, a single building will be counted in one of your RSU but not in the other...). Thus you would need to use a single circle in your RSU table and then iterate for all your circles. But even with this method, you would have an overestimation of some indicators since the full building will be considered for facade density calculation for example, even though only a small part of it is part of the RSU. It means that bigger is your circle, the more accurate the method...
2.  To solve the issue previously described, it is now possible to use specific algorithm (coded in GeoClimate) to calculate the exact facade area contained in a given RSU. However, they are not automatically used. Thus it would mean to call the function to calculate each of the indicator which might be impacted by the previously discussed problem:
    - SVF: fine if you do not use the simplified approach
    - building fraction: fine
    - aspect ratio: need as input the facade density, need to use the "exact version" to calculate it
    - height of roughness elements: fine
    - terrain roughness class: derived from the roughness length, which is calculated using as input the projected_facade_area_distribution which should be the "exact version"

Concerning the groovy work that is needed to make it work, the first is not quite complicated (I think we may provide you a working code before May the 17th. Concerning the second it is more complicated I cannot say we would have it before the end of your work. Except if @ebocher is super hot  =)

**Reply by Jukjuk-boop** [reply]
Hi @j3r3m1, thank you for the quick reply.

For my study I think the first option will be fine enough, the goal is to identify how we should consider LCZ to evaluate their relations with temperature. Therefore, I think it should be enough to give us an hint if this is the right way.

If I understood correctly, right now there's nothing I can do to help get this task done ?

The second method looks really nice, I would like to give it a try.
But don't rush it, maybe I will get the chance to use this one next year if I keep doing researches !

**Comment by j3r3m1** [comment]
Here is the code for the 1st solution: https://github.com/j3r3m1/LczForStationBuffer
However, it does not work yet (cf. https://github.com/orbisgis/geoclimate/issues/966).

If you want to test it on your computer, feel free to download the whole content of the project, install IntelliJ and open the project in IntelliJ. Then you can use the data which is given in the issue mentionned above. You will have to slightly modify the code to select the folder at the right location on your computer.

**Comment by j3r3m1** [comment]
Should now work, I have added the minimum information about "how to run the code" on the main page of the repo. 

@Jukjuk-boop 
Feel free to ask for more information for any of the steps

**Comment by j3r3m1** [comment]
@Jukjuk-boop, feel free to mark the discussion as answered if now fine with you
